packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -46,24 +60,90 @@ const useAnimalOrBatchRemoval = (
           removal_explanation: formData.explanation,
           removal_date: timestampedDate,
         });
+        selectedAnimalIds.push(id);
       } else if (kind === AnimalOrBatchKeys.BATCH) {
         animalBatchRemovalArray.push({
           id: entity_id,
           animal_removal_reason_id: Number(formData.reason),
           removal_explanation: formData.explanation,
           removal_date: timestampedDate,
         });
+        selectedBatchIds.push(id);
+      }
+    }
+
+    try {
+      if (animalRemovalArray.length) {
+        await mutations['removeAnimals'].trigger(animalRemovalArray).unwrap();
+        setSelectedInventoryIds(selectedInventoryIds.filter((i) => !selectedAnimalIds.includes(i)));
+        dispatch(enqueueSuccessSnackbar(t('ANIMALS.SUCCESS_REMOVE_ANIMALS', { ns: 'message' })));
+      }
+    } catch (err) {
+      dispatch(enqueueErrorSnackbar(t('ANIMALS.FAILED_REMOVE_ANIMALS', { ns: 'message' })));
+    }
+
+    try {
+      if (animalBatchRemovalArray.length) {
+        await mutations['removeBatches'].trigger(animalBatchRemovalArray).unwrap();
+        setSelectedInventoryIds(selectedInventoryIds.filter((i) => !selectedBatchIds.includes(i)));
+        dispatch(enqueueSuccessSnackbar(t('ANIMALS.SUCCESS_REMOVE_BATCHES', { ns: 'message' })));
+      }
+    } catch (err) {
+      dispatch(enqueueErrorSnackbar(t('ANIMALS.FAILED_REMOVE_BATCHES', { ns: 'message' })));
+    }
+
+    setRemovalModalOpen(false);
+  };
+
+  const handleAnimalOrBatchDeletion = async () => {
+    const animalIds: number[] = [];
+    const selectedAnimalIds: string[] = [];
+    const animalBatchIds: number[] = [];
+    const selectedBatchIds: string[] = [];
+
+    for (const id of selectedInventoryIds) {
+      const { kind, id: entity_id } = parseInventoryId(id);
+      if (kind === 'ANIMAL') {
```

Comment:
I asked Joyce to replace this with the enum `AnimalOrBatchKeys`! Could you please do the same?

Reply:
Yeah for sure!